### ISSUE TITLE
[ALM Accelerator - BUG]: No organization is matching the given Dataset - SerchDeploymentEnvironnementTeams

### ISSUE THREAD

**Issue by tocri** [microsoft/coe-starter-kit#7948]
### Describe the issue

 have a licenced user who has this notify when he clicks on configure deployment settings for a solution.

![image](https://github.com/microsoft/coe-starter-kit/assets/10110260/e266e1cd-55ad-46cf-bede-2a1195166639)

He has the SR Environnement Maker on the dev envrionnement.
He can commit and deploy the solution but can't go into custom environnemet settings

If i open the flow i have

![image](https://github.com/microsoft/coe-starter-kit/assets/10110260/d2e14418-84d9-4360-880d-e6d54541d4f1)

![image](https://github.com/microsoft/coe-starter-kit/assets/10110260/0139bf3c-9ed1-4ef0-83d5-05cd26a37e34)

it seems to come from an environment that no longer exists. It is important to specify that this error does not concern all developers. And each failed iteration only concerns this environment.

unqbaa868870d3f475eb9df096eedfc7.crm3

the orginal post : https://github.com/microsoft/coe-starter-kit/issues/7921

### What component are you experiencing the issue with?

ALM Accelerator Canvas App

### What solution version are you using?

1.0.20240305.1

### Steps To Reproduce

1. Click on configure deployment settings
2

AB#2679

**Comment by mikefactorial** [comment]
@tocri is there a deployment step in the profile they are using that is referencing the since deleted environment?

**Comment by tocri** [comment]
thanks for the reply. It will be to easy if it was the origin of the issue :-).
The profile deployment step is correct. Nothing has changed. That's weird because it's impact only few developpers. There is no issue with the others

**Comment by tocri** [comment]
Correct me if i'm wrong. A dev just needs the SR basic User on test environnement and production environnement do deploy with ALM ?

**Comment by mikefactorial** [comment]
@tocri yes, they should only need Basic User. I assume the user in question is able to access the environment otherwise from the maker portal? If so, it's difficult to tell why it would be an issue for some users and not others unless there is some sort of difference between the users that you can identify. Is it possible that the Security Group for the environment doesn't include the user for example?

**Comment by tocri** [comment]
So I found out yesterday that IT has assigned the PowerPlatform Administrator role to a security group that I have no control over. On the one hand, I deleted the SR Admin System for the user from the new COE manage permission apps, thinking that there had been a bug, and then I gave the basic user permission to do so manually.

As such, this could explain the app's weird behavior because the people I'm talking about are part of the group mentioned.

I'm validating with IT to remove the Power Platform Administrator role from this group.
